src/Shouldly/ShouldlyEnvironmentContext.cs

```diff
@@ -0,0 +1,28 @@
+namespace Shouldly
+{
+    #if NETSTANDARD 
+    internal static class ShouldlyEnvironmentContext
+    {
+        public static bool IsWindows()
+           => System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Windows);
+
+        public static bool IsMac()
+            => System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.OSX);
+        
+        public static bool IsLinux()
+           => System.Runtime.InteropServices.RuntimeInformation.IsOSPlatform(System.Runtime.InteropServices.OSPlatform.Linux);
+    }
+    #else
+    internal static class ShouldlyEnvironmentContext
+    {
+        public static bool IsWindows()
+            => System.Environment.OSVersion.Platform == System.PlatformID.Win32NT;
+
+        public static bool IsMac()
+            => System.Environment.OSVersion.Platform == System.PlatformID.MacOSX;
```

Comment:
On mono apparently this won't work and it will report as unix. Not sure there's much we can about this

Reply:
I saw something about that, I figure I'll wait to see if anyone complains 😨